pennylane/workflow/jacobian_products.py

```diff
@@ -741,10 +743,18 @@ def compute_vjp(self, tapes, dy):
                 raise NotImplementedError(
                     "Lightning device VJPs are not supported with jax jacobians."
                 )
-            vjp_f = self._device.vjp(
-                numpy_tape.measurements, dyi, **self._processed_gradient_kwargs
-            )
-            out = vjp_f(numpy_tape)
+
+            if self._uses_new_device:
+                if self._execution_config is not None:
+                    out = self._device.compute_vjp(numpy_tape, dyi, self._execution_config)
+                else:
+                    out = self._device.compute_vjp(numpy_tape, dyi)
```

Comment:
I understand that we need to separate out the execution kwarg if it is `None`, in order to not override the `DefaultExecutionConfig` in `_device.compute_vjp`. However, I'm wondering whether 
1. this is the most stable API for `compute_vjp`, or whether maybe the `DefaultExecutionConfig` should be passed down by default, confirming the default instead of creating a separate logical branch for it, and
2. there is a reason that we do not to do this for DQ? :thinking: Maybe I'm missing it and we actually are doing it, but I did not find it.

Reply:
The `DeviceJacobianProducts` class only works with devices using the new API, hence includes the `ExecutionConfig` as a required positional argument. However, `LightningVJPs` is supposed to work with both the old and new device API, since LGPU and LKokkos still use the old API, which does not work with `ExecutionConfig`s. I think it makes more sense to leave this as is for now, and then we can remove these branches, or maybe even this class altogether once LGPU and LKokkos are updated as well.

Reply:
Had some discussion with @albi3ro offline. Execution has been updated to use `DeviceJacobianProducts` with `lightning.qubit`.